### DISCUSSION TITLE
How to setup GitHub action in a self hosted VM instance ?

### DISCUSSION THREAD

**Discussion by jyotirbhandari** [community/community#25553] [Actions]
<p>Hi,</p>
<p>We are struggling to identify our CI/CD platform, our repositories reside in GitHub and we are using third party CI/CD to deploy. We want to run CI/CD internally in our hosted environment and want to setup GitHub actions internally on self hosted instances for security reasons. </p>
<p>There is no proper documentation as to how to deploy GitHub actions on VM in self hosted environment and integrate as an when code is pushed or merged. It is mentioned that you need to register actions, which is also not specified as to how to register it in repositories.</p>
<p>Hope to get a response from your end.</p>
<p>Regards,</p>
<p>Jyotir</p>

**Comment by ethomson** [comment, marked as answer]
<p>Hello, Jyotir, I’m afraid this capability doesn’t exist yet.</p>
<p>We’re working on this functionality now and it will be available within the next few months, before we enter “GA” for GitHub Actions.</p>

**Comment by seidtgeist** [comment]
<p>Any updates on self-hosted runner availability and pricing?</p>

**Comment by dynjo** [comment]
<p>Any more timeline updates here, we are ready to switch but this piece will be critical.</p>

**Comment by koteswara-kelam** [comment]
<p>Follow this to add self-hosted runner to github: <a href="https://github.blog/2019-11-05-self-hosted-runners-for-github-actions-is-now-in-beta/" rel="nofollow noopener">https://github.blog/2019-11-05-self-hosted-runners-for-github-actions-is-now-in-beta/</a></p>
